File: Ruby/printer.rb
require "./types"

def pr_str(mal)
    if mal.kind_of?(Array) then
        res = mal.each {|x| pr_str(x)}
        res2 = res.join(" ")
        return "(#{res2})"
    end

    if mal.kind_of?(MalSymbol) then
        return mal.to_s
    end

    if mal.kind_of?(MalNumber) then
        return mal.to_s
    end
end

File: Python/printer.py
from mal_types import MalNumber, MalString, MalSymbol, MalError

def pr_str(mal):
    match type(mal).__name__:
        case list.__name__:
            res = map(pr_str, mal)
            return '(' + ' '.join(res) + ')'
        case MalNumber.__name__:
            return str(mal.val)
        case MalString.__name__:
            return str(mal.val)
        case MalSymbol.__name__:
            return str(mal.val)
        case MalError.__name__:
            return str(mal.val)
